File: src/main/java/io/github/takenoha/towerdefense/persistence/TeamInvitationMutationResult.java
package io.github.takenoha.towerdefense.persistence;

import java.util.Objects;
import java.util.Optional;

/** Result of an idempotent team-invitation mutation. */
public record TeamInvitationMutationResult(
        ManagementOutcome outcome,
        TeamInvitation invitation,
        Optional<TeamRecord> team) {
    public TeamInvitationMutationResult {
        Objects.requireNonNull(outcome, "outcome");
        Objects.requireNonNull(invitation, "invitation");
        Objects.requireNonNull(team, "team");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/persistence/TeamInvitationMutationResult.kt
package io.github.takenoha.towerdefense.persistence

import java.util.Optional
import kotlin.jvm.JvmRecord

/** Result of an idempotent team-invitation mutation. */
@JvmRecord
data class TeamInvitationMutationResult(
    val outcome: ManagementOutcome,
    val invitation: TeamInvitation,
    val team: Optional<TeamRecord>,
)
